### ISSUE TITLE
extra_arguments with -var

### ISSUE THREAD

**Issue by dmrzzz** [gruntwork-io/terragrunt#140]
I want to pass an individual variable to Terraform using `-var`.  The following seems like it should work, but doesn't (with terragrunt 0.10.2).

terraform.tfvars:
```hcl
terragrunt = {
  lock = {
    backend = "dynamodb"
    config {
      aws_region = "us-east-2"
      state_file_id = "dmrztest/${path_relative_to_include()}"
    }
  }

  remote_state = {
    backend = "s3"
    config {
      region = "us-east-2"
      encrypt = "true"
      bucket = "terraform.uiuc-tech-services-sandbox.aws.illinois.edu" #FIXME
      key = "dmrztest/${path_relative_to_include()}/terraform.tfstate"
    }
  }

  # Pass "bucket" as an input variable to Terraform
  terraform {
    extra_arguments "bucket" {
      arguments = [
        "-var 'bucket=terraform.uiuc-tech-services-sandbox.aws.illinois.edu'" #FIXME
      ]
      commands = [
        "apply",
        "plan",
        "import",
        "push",
        "refresh"
      ]
    }
  }
}
```
Note: no .tf files at all are required to reproduce the problem; terraform.tfvars is the only thing in the directory.

```bash
$ terragrunt plan
[terragrunt] 2017/02/20 18:13:27 Reading Terragrunt config file at /home/dmrz/tmp/tgtest/terraform.tfvars
[terragrunt] 2017/02/20 18:13:27 Initializing remote state for the s3 backend
[terragrunt] 2017/02/20 18:13:28 Configuring remote state for the s3 backend
[terragrunt] 2017/02/20 18:13:28 Running command: /usr/local/libexec/terraform remote config -backend s3 -backend-config=region=us-east-2 -backend-config=encrypt=true -backend-config=bucket=terraform.uiuc-tech-services-sandbox.aws.illinois.edu -backend-config=key=dmrztest/./terraform.tfstate
Initialized blank state with remote state enabled!
Remote state configured and pulled.
[terragrunt] 2017/02/20 18:13:30 Running command: /usr/local/libexec/terraform plan -var 'bucket=terraform.uiuc-tech-services-sandbox.aws.illinois.edu'
flag provided but not defined: -var 'bucket
Usage: terraform plan [options] [DIR-OR-PLAN]

  Generates an execution plan for Terraform.

  This execution plan can be reviewed prior to running apply to get a
  sense for what Terraform will do. Optionally, the plan can be saved to
  a Terraform plan file, and apply can take this plan file to execute
  this plan exactly.

  If a saved plan is passed as an argument, this command will output
  the saved plan contents. It will not modify the given plan.

Options:
[...snip...]

  -var 'foo=bar'      Set a variable in the Terraform configuration. This
                      flag can be set multiple times.

  -var-file=foo       Set variables in the Terraform configuration from
                      a file. If "terraform.tfvars" is present, it will be
                      automatically loaded if this flag is not specified.
exit status 1
[terragrunt] 2017/02/20 18:13:30 exit status 1
```

If I actually run the command it claims to be running, the syntax is fine: 
```bash
$ /usr/local/libexec/terraform plan -var 'bucket=terraform.uiuc-tech-services-sandbox.aws.illinois.edu'
Error loading config: No Terraform configuration files found in directory: /home/dmrz/tmp/tgtest
```
but it looks like Terragrunt is not recognizing the space and is trying to pass an option whose name is `-var 'bucket`

**Comment by dmrzzz** [comment]
OK, I eventually figured out the solution on my own:
```hcl
  # Pass "bucket" as an input variable to Terraform
  terraform {
    extra_arguments "bucket" {
      arguments = [
        "-var", "bucket=terraform.uiuc-tech-services-sandbox.aws.illinois.edu"
      ]
      commands = [
        "apply",
        "plan",
        "import",
        "push",
        "refresh"
      ]
    }
  }
```
but this was sufficiently non-obvious that I would recommend adding an example to the README.

Note in particular that it's important _not_ to include single quotes inside the second argument.

**Comment by brikis98** [comment]
> but this was sufficiently non-obvious that I would recommend adding an example to the README.

Agreed! Would you like to submit a PR to make this clearer?

**Comment by dmrzzz** [comment]
Whoops, sorry for not replying sooner; I'm afraid I don't have any time to spend on this right now.  At least the answer is here for anybody else who comes looking.

**Comment by kwilczynski** [comment]
@dmrzzz thank you so much for reporting this. Since you are busy, I took the liberty of sending a small Pull Request (see: https://github.com/gruntwork-io/terragrunt/pull/165) to clarify the whitespaces usage in the `arguments` attribute of `extra_arguments`.

**Comment by brikis98** [comment]
Fixed via #165.
